ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -1710,6 +1709,32 @@ public void shouldThrowServerErrorOnFailedToDistribute() {
     makeRequest(statement);
   }
 
+  @Test
+  public void shouldUseOneServiceContextPerKsqlRequest() {
+    // Given:
+    final ServiceContext sc = spy(TestServiceContext.create());
```

Comment:
let's avoid spies if possible! this leads me to two questions:
1. why do we need a spy instead of a mock? can you mock whatever is necessary or use a simpler KSQL statement list?
2. I'm not sure I see the connection between the test title (use one per request) and what we're actually asserting (was closed once). What if I created 10 different service contexts during that time and just closed one of them?

Reply:
Aa, you got the same question I had after submitting this patch. I was thinking on mocking the ServiceContext supplier and the ServiceContext itself, and keep track of the # of times a `get()` and `close` get called. Would that be a good approach?

Reply:
I like that approach :D

Reply:
I found a way to do it.
done.